armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -534,7 +485,8 @@ class ExpansionData:
 
     def __init__(self, a, setFuel):
         self._a = a
-        self._oldHotTemp = {}
+        self.componentReferenceHeight = {}
+        self.componentReferenceTemperature = {}
```

Comment:
1. Are these two dictionaries filled out during every run, or just during Axial Expansion runs?
2. How safe are they?  Could the height of these blocks get changed due to a non-axial-thermal-expansion method, during a run, and invalidate these dicts?

Reply:
> Are these two dictionaries filled out during every run, or just during Axial Expansion runs?

They are initialized and filled out when an axial expansion instance is created. So only during axial expansion runs. 

> Could the height of these blocks get changed due to a non-axial-thermal-expansion method, during a run, and invalidate these dicts?

The heights and temps that get stored in these dicts can certainly change. And that's expected. 

The class `ExpansionData` is only used within the `AxialExpansionChanger` class; so when an instance of the axial expansion changer goes out of scope, so does the expansion data instance (at least that's my intent). 

These dicts are designed to store the "previous" heights and temps (usually from a previous time/burn step or iteration from a coupled solve). So if you were to iterate on thermal hydraulics, fuel performance (e.g., burnup-based expansion), and neutronics for a coupled calculation, every time the axial expansion changer would fire up the "old" temps and heights would get stored here.

Reply:
So, is it true to say that after Axial Expansion is done, these dicts get deleted?

The situation we don't want is if these dicts were to persist between cycles in a long simulation.

We've seen this before, if you have a reference to every Block/Component/Whatever in the Reactor that persists after a Cycle is done, then you have a memory leak, and memory will just slowly go up over time. Does that make sense?